### PR TITLE
Remote compressed patches are broken

### PR DESCRIPTION
**Pull request by junghans**
This feature is only used in `nwchem@6.6`, but the patched there are gone.

### CODE REVIEW

File: var/spack/repos/builtin/packages/gsl/package.py [hunk 1]
```diff
@@ -24,3 +24,33 @@ class Gsl(AutotoolsPackage, GNUMirrorPackage):
     version('2.1',   sha256='59ad06837397617f698975c494fe7b2b698739a59e2fcf830b776428938a0c66')
     version('2.0',   sha256='e361f0b19199b5e6c21922e9f16adf7eca8dd860842802424906d0f83485ca2d')
     version('1.16',  sha256='73bc2f51b90d2a780e6d266d43e487b3dbd78945dd0b04b14ca5980fe28d2f53')
+
+    variant('external-cblas', default=False, description='Build against external blas')
+
+    #patch('https://dev.gentoo.org/~mgorny/dist/gsl-2.3-cblas.patch.bz2',
```
Comment (junghans) [inline var/spack/repos/builtin/packages/gsl/package.py]: @tgamblin @scheibelp @adamjstewart I have not idea why compressed patches are broken again! This feature is only used in `nwchem@6.6`, but the patches there are gone.

**Comment by junghans** [comment]
Made some progress, now the error is:
```
=> Error: NoArchiveFileError: Cannot call archive() before fetching.

/home/junghans/spack.git/lib/spack/spack/package.py:1123, in do_fetch:
       1120
       1121        for patch in self.spec.patches:
       1122            patch.fetch()
  >>   1123            if patch.cache():
       1124                patch.cache().cache_local()

```

@alalazo any idea?

**Comment by junghans** [comment]
I made a version that works around this issue by bundling it in #15261.

**Comment by alalazo** [comment]
@junghans Has this been fixed by #15261 ?

**Comment by junghans** [comment]
Well remote patches are still broken, I think.

**Comment by alalazo** [comment]
@junghans Can you rebase this one?

**Comment by junghans** [comment]
Outdated.

**Pull request closed without merging**